src/measure_theory/integral/circle_integral.lean

```diff
@@ -0,0 +1,513 @@
+by simp [circle_map, exp_ne_zero]
+
+@[simp] lemma circle_map_zero_radius (c : ℂ) : circle_map c 0 = const ℝ c :=
+funext $ λ θ, circle_map_eq_center_iff.2 rfl
+
+lemma circle_map_ne_center {c : ℂ} {R : ℝ} (hR : R ≠ 0) {θ : ℝ} : circle_map c R θ ≠ c :=
+mt circle_map_eq_center_iff.1 hR
+
+lemma has_deriv_at_circle_map (c : ℂ) (R : ℝ) (θ : ℝ) :
+  has_deriv_at (circle_map c R) (circle_map 0 R θ * I) θ :=
+by simpa only [mul_assoc, one_mul, of_real_clm_apply, circle_map, of_real_one, zero_add]
+ using ((of_real_clm.has_deriv_at.mul_const I).cexp_real.const_mul (R : ℂ)).const_add c
+
+lemma differentiable_circle_map (c : ℂ) (R : ℝ) :
+  differentiable ℝ (circle_map c R) :=
+λ θ, (has_deriv_at_circle_map c R θ).differentiable_at
+
+@[continuity] lemma continuous_circle_map (c : ℂ) (R : ℝ) : continuous (circle_map c R) :=
+(differentiable_circle_map c R).continuous
+
+@[measurability] lemma measurable_circle_map (c : ℂ) (R : ℝ) : measurable (circle_map c R) :=
+(continuous_circle_map c R).measurable
+
+@[simp] lemma deriv_circle_map (c : ℂ) (R : ℝ) (θ : ℝ) :
+  deriv (circle_map c R) θ = circle_map 0 R θ * I :=
+(has_deriv_at_circle_map _ _ _).deriv
+
+lemma deriv_circle_map_eq_zero_iff {c : ℂ} {R : ℝ} {θ : ℝ} :
+  deriv (circle_map c R) θ = 0 ↔ R = 0 :=
+by simp [I_ne_zero]
+
+lemma deriv_circle_map_ne_zero {c : ℂ} {R : ℝ} {θ : ℝ} (hR : R ≠ 0) :
+  deriv (circle_map c R) θ ≠ 0 :=
+mt deriv_circle_map_eq_zero_iff.1 hR
+
+lemma lipschitz_with_circle_map (c : ℂ) (R : ℝ) :
+  lipschitz_with R.nnabs (circle_map c R) :=
+lipschitz_with_of_nnnorm_deriv_le (differentiable_circle_map _ _) $ λ θ,
+  nnreal.coe_le_coe.1 $ by simp
+
+/-!
+### Integrability of a function on a circle
+-/
+
+/-- We say that a function `f : ℂ → E` is integrable on the circle with center `c` and radius `R` if
+the function `f ∘ circle_map c R` is integrable on `[0, 2π]`.
+
+Note that the actual function used in the definition of `circle_integral` is
+`(deriv (circle_map c R) θ) • f (circle_map c R θ)`. Integrability of this function is equivalent
+to integrability of `f ∘ circle_map c R` whenever `R ≠ 0`. -/
+def circle_integrable (f : ℂ → E) (c : ℂ) (R : ℝ) : Prop :=
+interval_integrable (λ θ : ℝ, f (circle_map c R θ)) volume 0 (2 * π)
+
+@[simp] lemma circle_integrable_const (a : E) (c : ℂ) (R : ℝ) :
+  circle_integrable (λ _, a) c R :=
+interval_integrable_const
+
+namespace circle_integrable
+
+variables {f g : ℂ → E} {c : ℂ} {R : ℝ}
+
+lemma add [borel_space E] [second_countable_topology E]
+  (hf : circle_integrable f c R) (hg : circle_integrable g c R) :
+  circle_integrable (f + g) c R :=
+hf.add hg
+
+lemma neg [borel_space E] (hf : circle_integrable f c R) : circle_integrable (-f) c R := hf.neg
+
+/-- The function we actually integrate over `[0, 2π]` in the definition of `circle_integral` is
+integrable. -/
+lemma out [borel_space E] [normed_space ℂ E] [second_countable_topology E]
```

Comment:
Why this name?

Reply:
Because I failed to come up with a meaningful name.